src/Paprika/Utils/RefCountingDisposable.cs

```diff
@@ -11,13 +12,12 @@ public abstract class RefCountingDisposable : IDisposable
     private const int Initial = 1;
     private const int NoAccessors = 0;
     private const int Disposing = -1;
-    private const int DisposingBarrier = 0;
 
-    private int _leases;
+    private PaddedValue _leases;
```

Comment:
I see the padding here. I know what it does. The thing is that this is the abstract class and what I'm thinking is whether is needed. WDYT?

Reply:
If it is hevily used changing its value causes anything on its cache line to be invalidated for other CPUs and forces them to be reloaded.

So for example the pointer of `HashSet<ulong> _bloom` would keep getting invaliated in `BlockState`; or pointer of `PooledSpanDictionary _dict` in `ChildCommit`

Reply:
Depends if worried about the memory usage by the extra padding or not